packages/rum-core/src/domain/view/viewMetrics/trackNavigationTimings.spec.ts

```diff
@@ -1,30 +1,71 @@
-import type { Duration } from '@datadog/browser-core'
+import type { Duration, RelativeTime } from '@datadog/browser-core'
+import type { Clock } from '@datadog/browser-core/test'
+import { mockClock, registerCleanupTask } from '@datadog/browser-core/test'
+import type { RumPerformanceEntry } from '../../../browser/performanceObservable'
 import { RumPerformanceEntryType } from '../../../browser/performanceObservable'
-import type { TestSetupBuilder } from '../../../../test'
-import { createPerformanceEntry, setup } from '../../../../test'
-import { LifeCycleEventType } from '../../lifeCycle'
+import { createPerformanceEntry, mockPerformanceObserver, mockPerformanceTiming } from '../../../../test'
+import type { RumConfiguration } from '../../configuration'
 import type { NavigationTimings } from './trackNavigationTimings'
 import { trackNavigationTimings } from './trackNavigationTimings'
 
 describe('trackNavigationTimings', () => {
-  let setupBuilder: TestSetupBuilder
   let navigationTimingsCallback: jasmine.Spy<(timings: NavigationTimings) => void>
+  let notifyPerformanceEntries: (entries: RumPerformanceEntry[]) => void
+  let stop: () => void
+  let clock: Clock
+
+  function removePerformanceObserver() {
+    const originalPerformanceObserver = window.PerformanceObserver
+    window.PerformanceObserver = undefined as any
+
+    registerCleanupTask(() => {
+      window.PerformanceObserver = originalPerformanceObserver
+    })
+  }
 
   beforeEach(() => {
     navigationTimingsCallback = jasmine.createSpy()
+  })
 
-    setupBuilder = setup().beforeBuild(({ lifeCycle }) => trackNavigationTimings(lifeCycle, navigationTimingsCallback))
+  afterEach(() => {
```

Comment:
**🥜 nitpick:** ‏use registerCleanupTask

Reply:
You can also remove `clock?.cleanup()` if you're up to date with main